packages/widgets/src/blocks/legacy-widget/edit/index.js

```diff
@@ -32,14 +33,44 @@ import ConvertToBlocksButton from './convert-to-blocks-button';
 
 export default function Edit( props ) {
 	const { id, idBase } = props.attributes;
-	const { isWide = false } = props;
+
+	const [ isEditing, setIsEditing ] = useState( false );
+	const stopEditing = useCallback( () => {
+		setIsEditing( false );
+		ref.current.focus();
+	}, [ setIsEditing ] );
+
+	props = { ...props, isEditing, stopEditing };
+
+	// isWide is applicable only at breakpoints greater than small
+	const canWideFit = useViewportMatch( 'small' );
+	props.isWide &&= canWideFit;
```

Comment:
I don't think mutating props is good practice. Ideally these become new constants.

Reply:
✔️